### ISSUE TITLE
Upcast check is missing a null check when casting to a nullable type

### ISSUE THREAD

**Issue by dcodeIO** [AssemblyScript/assemblyscript#2035]
Code like

```ts
class Foo {}
class Bar extends Foo {}
var foo: Foo | null = null;
foo instanceof Bar | null // here
```

is missing a check for when the LHS value is `0`, directly calling the `__instanceof` helper which then can fail due to UB.

**Comment by trusktr** [comment]
Related: https://github.com/AssemblyScript/assemblyscript/issues/1976

**Comment by trusktr** [comment]
Does this fix #1976 ? I haven't circled back to that yet.

**Comment by dcodeIO** [comment]
If I see correctly, the issue is also an upcast check involving `null`, so yeah, it think it should.
